### PR TITLE
JN-762: renaming "notificationConfig" to "trigger"

### PR DESCRIPTION
**Pull request by devonbush**
#### DESCRIPTION

This renames "NotificationConfig" to "Trigger" in what I think is a more accurate name of what it's doing now, and also the direction we want to move it in.  Other than some trivial UI text updates, this PR has no functional changes.

@connorlbark  this will massively merge conflict with your PR, so let's make sure yours gets merged in first, and then I'll update this one to match

#### TO TEST:  *(simple manual steps for confirming core behavior -- used for pre-release checks)*
1. restart ApiAdminApp
2. go to https://localhost:3000/ourhealth/studies/ourheart/env/sandbox/notificationContent
3. see that the notifications still appear and are editable, etc...

### CODE REVIEW

File: api-admin/src/main/java/bio/terra/pearl/api/admin/controller/notifications/TriggerController.java [hunk 1]
```diff
@@ -61,21 +67,55 @@ public ResponseEntity<Object> replace(
...
+        studyShortcode,
+        EnvironmentName.valueOfCaseInsensitive(envName),
+        configId,
+        enrolleeRuleData);
+    return ResponseEntity.ok().build();
+  }
+
+  /** send a one-off notification. */
+  @Override
+  public ResponseEntity<Object> adHoc(
```
Comment (connorlbark) [inline api-admin/src/main/java/bio/terra/pearl/api/admin/controller/notifications/TriggerController.java:94]: I like this change - moving adHoc/test to the triggercontroller makes a lot more sense. But should we also move `sendAdHoc` to the trigger service?

File: api-admin/src/main/resources/api/openapi.yml [hunk 2]
```diff
@@ -721,24 +721,24 @@ paths:
...
         '200':
-          description: The notification configs
+          description: The triggered actions
           content: { application/json: { schema: { type: object } } }
         '500':
           $ref: '#/components/responses/ServerError'
     post:
-      summary: create a new notification config for the environment
-      tags: [ notificationConfig ]
+      summary: create a new triggered action for the environment
```
Comment (connorlbark) [inline api-admin/src/main/resources/api/openapi.yml:740]: Just curious - the wording "triggered action" makes me think it won't just be for notifications in the future. Is that true, or are triggers purely for notifications? if so, should we do work (in a new PR) to disentangle the connection between a `Trigger` and emails?

File: core/src/main/java/bio/terra/pearl/core/service/publishing/PortalPublishingService.java [hunk 3]
```diff
@@ -81,7 +81,7 @@ protected PortalEnvironment applyUpdate(PortalEnvironment destEnv, PortalEnviron
 
         applyChangesToPreRegSurvey(destEnv, envChanges.preRegSurveyChanges());
         applyChangesToSiteContent(destEnv, envChanges.siteContentChange());
-        applyChangesToNotificationConfigs(destEnv, envChanges.notificationConfigChanges());
+        applyChangesToNotificationConfigs(destEnv, envChanges.triggerChanges());
```
Comment (connorlbark) [inline core/src/main/java/bio/terra/pearl/core/service/publishing/PortalPublishingService.java]: `applyChangesToNotificationConfigs` should be updated to `applyChangesToTriggers`

File: core/src/main/java/bio/terra/pearl/core/service/publishing/StudyPublishingService.java [hunk 4]
```diff
@@ -122,23 +122,23 @@ private List<StudyEnvironmentSurvey> applyChangesToSurveys(StudyEnvironment dest
         return destEnv.getConfiguredSurveys();
     }
 
-    protected void applyChangesToNotificationConfigs(StudyEnvironment destEnv, ListChange<NotificationConfig,
+    protected void applyChangesToNotificationConfigs(StudyEnvironment destEnv, ListChange<Trigger,
```
Comment (connorlbark) [inline core/src/main/java/bio/terra/pearl/core/service/publishing/StudyPublishingService.java]: same here

File: populate/src/main/java/bio/terra/pearl/populate/service/EnrolleePopulator.java [hunk 5]
```diff
@@ -270,10 +270,10 @@ private String getTargetName(TaskType taskType, String stableId, int version) {
 
     private void populateNotifications(Enrollee enrollee, EnrolleePopDto enrolleeDto, UUID studyEnvironmentId,
     PortalParticipantUser ppUser) {
-        List<NotificationConfig> notificationConfigs = notificationConfigService.findByStudyEnvironmentId(studyEnvironmentId);
+        List<Trigger> triggers = triggerService.findByStudyEnvironmentId(studyEnvironmentId);
         for (NotificationPopDto notificationPopDto : enrolleeDto.getNotifications()) {
-            NotificationConfig matchedConfig = matchConfigToNotification(notificationConfigs, notificationPopDto);
-            notificationPopDto.setNotificationConfigId(matchedConfig.getId());
+            Trigger matchedConfig = matchConfigToNotification(triggers, notificationPopDto);
```
Comment (connorlbark) [inline populate/src/main/java/bio/terra/pearl/populate/service/EnrolleePopulator.java]: `matchConfigToNotification` should be `matchTriggerToNotification`

File: ui-admin/src/study/notifications/NotifcationConfigTypeDisplay.tsx [hunk 6]
```diff
@@ -15,17 +15,17 @@ export const eventTypeDisplayMap: Record<string, string> = {
 }
 
 /** shows a summary of the notification config */
-export default function NotificationConfigTypeDisplay({ config }: {config?: NotificationConfig}) {
+export default function NotificationConfigTypeDisplay({ config }: {config?: Trigger}) {
```
Comment (connorlbark) [inline ui-admin/src/study/notifications/NotifcationConfigTypeDisplay.tsx]: I noticed a lot of the UI components/variables still call them notification configs - are we keeping that way for continuity for users? if so, that might be confusing for new devs, but if not that might introduce a lot more changes in this PR

**Review by connorlbark — CHANGES_REQUESTED** [review]
great naming change! works great locally as well. Just some places I noticed still need renaming / clarifications

**Review by MatthewBemis — APPROVED** [review]

File: api-admin/src/main/java/bio/terra/pearl/api/admin/controller/notifications/TriggerController.java (hunk 1 above)
Comment (devonbush) [reply api-admin/src/main/java/bio/terra/pearl/api/admin/controller/notifications/TriggerController.java:94]: yeah, I waffled on that.  I think you're right it makes sense to move it

File: api-admin/src/main/resources/api/openapi.yml (hunk 2 above)
Comment (devonbush) [reply api-admin/src/main/resources/api/openapi.yml:740]: yeah, in the far future, we'll want to support triggering things that aren't notifications.  but that's a ways off

File: ui-admin/src/study/notifications/NotifcationConfigTypeDisplay.tsx (hunk 6 above)
Comment (devonbush) [reply ui-admin/src/study/notifications/NotifcationConfigTypeDisplay.tsx]: yeah, I tried to get most of them in this PR, but I wanted to get this in quickly, and then we can clean up the stragglers as we go

File: api-admin/src/main/java/bio/terra/pearl/api/admin/controller/notifications/TriggerController.java (hunk 1 above)
Comment (devonbush) [reply api-admin/src/main/java/bio/terra/pearl/api/admin/controller/notifications/TriggerController.java:94]: oh wait, now I know why I didn't move it -- that method has email-specific logic.  Eventually, when we have different types of triggers, this controller method might delegate to multiple different services depending on the trigger action specified.

File: ui-admin/src/study/notifications/NotifcationConfigTypeDisplay.tsx (hunk 6 above)
Comment (connorlbark) [reply ui-admin/src/study/notifications/NotifcationConfigTypeDisplay.tsx]: sounds good

**Review by connorlbark — APPROVED** [review]

**Pull request merged**